src/hpr_lorawan.erl

```diff
@@ -11,17 +11,83 @@
 
 -export([
     index_to_datarate/2,
-    datarate_to_index/2
+    datarate_to_index/2,
+    key_matches_mic/2
 ]).
 
 -type temp_datarate_index() :: non_neg_integer().
 
--spec index_to_datarate(atom(), temp_datarate_index()) -> atom().
+-spec index_to_datarate(Region :: atom(), DRIndex :: temp_datarate_index()) -> atom().
 index_to_datarate(Region, DRIndex) ->
     Plan = lora_plan:region_to_plan(Region),
     lora_plan:datarate_to_atom(Plan, DRIndex).
 
--spec datarate_to_index(atom(), atom()) -> integer().
+-spec datarate_to_index(Region :: atom(), DR :: atom()) -> integer().
 datarate_to_index(Region, DR) ->
     Plan = lora_plan:region_to_plan(Region),
     lora_plan:datarate_to_index(Plan, DR).
+
+-spec key_matches_mic(Key :: binary(), Payload :: binary()) -> boolean().
+key_matches_mic(Key, Payload) ->
+    ExpectedMIC = mic(Payload),
```

Comment:
Can we take this opportunity to export these functions in the erlang-lorawan lib?
https://github.com/helium/erlang-lorawan/blob/master/src/lora_core.erl#L16-L38